sdk/core/Azure.Core/samples/ProtocolMethods.md

```diff
@@ -0,0 +1,149 @@
+# C# Azure SDK Clients that Contain Protocol Methods
+
+## Introduction
+
+Most Azure SDK Clients expose methods that take 'model' parameters, C# classes which map to the request or response content of the REST call.
+
+Imagine a service that stores information about pets, with a GetDog and SetDog pair of operations. 
+
+Pets are represented in the request by a JSON object:
+
+```json
+{
+        "name": "Buddy",
+        "id": 2,
+        "color": "Brown"
+}
+```
+
+An API using models might be:
+
+```csharp
+        // This is an example model class
+        public class Pet
+        {
+                string Name { get; }
```

Comment:
these properties need to be public

Reply:
Are you sure? 

Example: https://github.com/Azure/autorest.csharp/blob/feature/v3/samples/CognitiveServices.TextAnalytics/Generated/Models/DetectedLanguage.cs#L37